packages/babel-plugin-proposal-dynamic-import/test/fixtures/amd/module/output.js

```diff
@@ -0,0 +1,5 @@
+define(["require"], function (_require) {
+  "use strict";
+
+  var modP = new Promise(_resolve => _require(["mod"], imported => _resolve(babelHelpers.interopRequireWildcard(imported))));
```

Comment:
I checked the require source, and apparentely the callback is always called after a `context.nextTick`:
https://github.com/requirejs/requirejs/blob/fdf4186d3e68df06a04bd71cb6ea0f24eb1600d1/require.js#L1460

It is always guaranteed to be async: https://github.com/requirejs/requirejs/issues/738#issuecomment-17708061

Reading RequireJS's source code made me realize that it can also generate an error.